rts/System/myMath.cpp

```diff
@@ -28,7 +28,10 @@ void CMyMath::Init()
 	}
 
 	unsigned checksum = 0;
+	// NOLINTNEXTLINE{modernize-loop-construct}
```

Comment:
1) the number of NOLINT's is already getting worrysome, source code should not be contorted to satisfy any single tool imo.

2) if float* and int* didn't have the same alignment, this code would have crashed and burned a long time ago.

Reply:
Wall of text but: 
1. The number of NOLINTs is a bit large, although most of them are trying to silence dead code `false && expr` or `true || expr`, and will probably go away once that section is fixed/implemented. 
However I agree with you on principle, and I'll try to get rid as many as I can by disabling some style checks globally, but I'd like to leave a few of them in place. I suppose it can serve as an extra `don't change this code` warning, but more importantly, it allows us to use clang-tidy with a 0 warnings setup. 
As long as it doesn't produce too many false positives, any tool that helps reduce the errors, increases performance and keeps code readable is important to use in my opinion.
In addition, I think C++ as a language is taking the path of annotating the code more to ensure safety, readability and to support tools issuing warning/error checks (https://en.cppreference.com/w/cpp/language/attributes#Standard_attributes). 

2. To the best of my understanding, this is UB that happens to be correct because of our compilers/target systems and maybe compile flags(not sure if we're using `-fno-strict-aliasing`). It seems to be a violation of strict aliasing rules: [[1]](https://stackoverflow.com/questions/98340/what-are-the-common-undefined-unspecified-behavior-for-c-that-you-run-into) [[2]](https://cellperformance.beyond3d.com/articles/2006/06/understanding-strict-aliasing.html)

Reply:
Those NOLINTs can go away if you want me to ignore [readability-simplify-boolean-expr](https://clang.llvm.org/extra/clang-tidy/checks/readability-simplify-boolean-expr.html) and [misc-static-assert](https://clang.llvm.org/extra/clang-tidy/checks/misc-static-assert.html) checks globally. Honestly I'd let them stay as those blocks of code *should* alert the user that something is odd (the actual check is disabled). Up to you.
PS: I'm thinking of replacing NOLINT(specific-check) with just NOLINT. It seems the whole // NOLINT  thing is used by other tools as well (e.g. [cpplint](https://en.wikipedia.org/wiki/Cpplint))

Reply:
Ignore them for now, I'll see which of these instances (probably most) can be removed or converted to comments later.